### PR TITLE
OCPBUGS-25025: go.mod: bump prometheus-operator to v0.70.0

### PR DESCRIPTION
**Pull request by jan--f**
* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by simonpasquier** [comment]
you need to update the vendor/ directory too.

**Comment by Tai-RedHat** [comment]
test with cluster-bot, able to see `trackTimestampsStaleness: true`
/label qe-approved

**Comment by Tai-RedHat** [comment]
The built cluster by this PR does not have `Observe` menu in the web console

File: go.mod [hunk 1]
```diff
@@ -98,54 +105,56 @@ require (
...
+	sigs.k8s.io/controller-runtime v0.16.3 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/kube-storage-version-migrator v0.0.6-0.20230721195810-5c8923c5ff96 // indirect
-	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
-	sigs.k8s.io/yaml v1.3.0 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
+	sigs.k8s.io/yaml v1.4.0 // indirect
 )
+
+replace k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20230928195430-ce36a0c3bb67
```
Comment (jan--f) [inline go.mod:160]: This is currently needed after https://github.com/kubernetes/kube-openapi/pull/431. I'm still trying to fogure out wherer to report this. I suspect `apiserver` needs to get a fix.

**Comment by jan--f** [comment]
> The built cluster by this PR does not have Observe menu in the web console

This might be unrelated to this PR but rather to https://issues.redhat.com/browse/OCPBUGS-23516?

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Pull request merged**